godel_scan_analysis/include/godel_scan_analysis/scan_algorithms.h

```diff
@@ -0,0 +1,144 @@
+#ifndef SCAN_ALGORITHMS_H
+#define SCAN_ALGORITHMS_H
+
+#include <algorithm>
+#include <numeric>
+
+#include "godel_scan_analysis/scan_utilities.h"
+
+namespace rms
+{
+
+/////////////////////////////////////////////////////////////////
+// Utility functions for fitting and using least-squares lines //
+/////////////////////////////////////////////////////////////////
+
+template<typename FloatType, typename InputIt>
```

Comment:
Could be float.

When I first went through and wrote this, I wanted it to be quick and flexible. The keyence can generate data at more than 1000 Hz and I wanted to keep up. Admittedly for our application, I doubt it will ever not be double but I wanted that flexibility to be in the mini-library.